From 411dccf9d271e49f37471c73ebedb18719d6b608 Mon Sep 17 00:00:00 2001
From: Minghao Chi <chi.minghao@zte.com.cn>
Date: Mon, 16 May 2022 11:54:12 +0000
Subject: [PATCH] dmaengine: idxd: Remove unnecessary synchronize_irq() before
 free_irq()
Git-commit: 411dccf9d271e49f37471c73ebedb18719d6b608
Patch-mainline: v5.19-rc1
References: bsn#314
Modified-by-SEL: Yes, refreshed due to different context


Calling synchronize_irq() right before free_irq() is quite useless. On one
hand the IRQ can easily fire again before free_irq() is entered, on the
other hand free_irq() itself calls synchronize_irq() internally (in a race
condition free way), before any state associated with the IRQ is freed.

Signed-off-by: Minghao Chi <chi.minghao@zte.com.cn>
Link: https://lore.kernel.org/r/20220516115412.1651772-1-chi.minghao@zte.com.cn
Acked-by: Dave Jiang <dave.jiang@intel.com>
Signed-off-by: Vinod Koul <vkoul@kernel.org>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/dma/idxd/device.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/dma/idxd/device.c
+++ b/drivers/dma/idxd/device.c
@@ -1176,7 +1176,6 @@ void idxd_wq_free_irq(struct idxd_wq *wq
 	struct idxd_device *idxd = wq->idxd;
 	struct idxd_irq_entry *ie = &wq->ie;
 
-	synchronize_irq(ie->vector);
 	free_irq(ie->vector, ie);
 	idxd_flush_pending_descs(ie);
 	if (idxd->request_int_handles)
